src/core/dbus-unit.c

```diff
@@ -1877,11 +1877,24 @@ int bus_unit_queue_job(
             (type == JOB_RELOAD_OR_START && job_type_collapse(type, u) == JOB_START && u->refuse_manual_start))
                 return sd_bus_error_setf(error, BUS_ERROR_ONLY_BY_DEPENDENCY, "Operation refused, unit %s may be requested by dependency only (it is configured to refuse manual start/stop).", u->id);
 
-        /* dbus-broker issues StartUnit for activation requests, so let's apply the same check
-         * used in signal_activation_request(). */
-        if (type == JOB_START && u->type == UNIT_SERVICE &&
-            SERVICE(u)->type == SERVICE_DBUS && !manager_dbus_is_running(u->manager))
-                return sd_bus_error_set(error, BUS_ERROR_SHUTTING_DOWN, "Refusing activation, D-Bus is not running.");
+        if (type == JOB_START && u->type == UNIT_SERVICE && SERVICE(u)->type == SERVICE_DBUS) {
+                /* dbus-broker issues StartUnit for activation requests, and Type=dbus services automatically
+                 * gain dependency on dbus.socket. Therefore, if dbus has a pending stop job, the new start
+                 * job that pulls in dbus again would result in job type conflict. */
```

Comment:
Please comment why `manager_dbus_is_running()` or `unit_inactive_or_pending()` cannot be used.